tiledb/api/src/array/mod.rs

```diff
@@ -1019,4 +1130,102 @@ pub mod tests {
 
         test_uri.close().map_err(|e| Error::Other(e.to_string()))
     }
+
+    fn create_simple_dense(
+        test_uri: &dyn TestArrayUri,
+        ctx: &Context,
+    ) -> TileDBResult<String> {
+        let domain = {
+            let rows = DimensionBuilder::new(
+                ctx,
+                "id",
+                Datatype::Int32,
+                ([1, 410], 10),
+            )?
+            .build();
+
+            DomainBuilder::new(ctx)?.add_dimension(rows)?.build()
+        };
+
+        let schema = SchemaBuilder::new(ctx, ArrayType::Dense, domain)?
+            .add_attribute(
+                AttributeBuilder::new(ctx, "a", Datatype::Int32)?.build(),
+            )?
+            .build()?;
+
+        let array_uri = test_uri
+            .with_path("quickstart_dense")
+            .map_err(|e| Error::Other(e.to_string()))?;
+        Array::create(ctx, &array_uri, schema)?;
+        Ok(array_uri)
+    }
+
+    fn write_dense_vector_4_fragments(
+        ctx: &Context,
+        array_uri: &str,
+        timestamp: u64,
+    ) -> TileDBResult<()> {
+        // Subarray boundaries.
+        let boundaries = [0, 200, 250, 310, 410];
+
+        for i in 0..4 {
+            // Prepare cell buffer.
+            let low_bound = boundaries[i];
+            let high_bound = boundaries[i + 1];
+
+            let mut data: Vec<i32> = vec![];
```

Comment:
`(low_bound.. high_bound).collect::<Vec<i32>>()` is more idiomatic

(and, not that it matters here but might some other place, will reserve space at the beginning to do one allocation)

Reply:
Thanks for the suggestion. I can also add that I can let the data be immutable.